lib/web_ui/flutter_js/BUILD.gn

```diff
@@ -2,7 +2,25 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-copy("flutter_js") {
-  sources = [ "flutter.js" ]
-  outputs = [ "$root_out_dir/flutter_web_sdk/{{source_file_part}}" ]
+import("//flutter/build/esbuild/esbuild.gni")
+import("sources.gni")
+
+group("flutter_js") {
+  public_deps = [
+    ":flutter_js_bundle",
+    ":flutter_js_sources",
+  ]
+}
+
+copy("flutter_js_sources") {
+  sources = flutter_js_source_list
+  outputs =
+      [ "$root_out_dir/flutter_web_sdk/flutter_js/{{source_target_relative}}" ]
+}
+
+esbuild("flutter_js_bundle") {
+  entry_point = "$root_out_dir/flutter_web_sdk/flutter_js/src/flutter.js"
+  output_bundle = "$root_out_dir/flutter_web_sdk/flutter_js"
```

Comment:
do we want this to be `$root_out_dir/flutter_web_sdk/flutter_js/dist` or similar? (this might be problematic for the sourcemaps maybe?)

Reply:
Yeah that will end up with the sourcemaps mapping to `../src/flutter.js`, which I don't think is going to work well when served via http. I'm not exactly sure how to around that issue with a `dist` folder to be honest.